include/string_span.h

```diff
@@ -178,4 +213,15 @@ template <size_t Max = dynamic_range>
 using wzstring_builder = basic_zstring_builder<wchar_t, Max>;
 }
 
+// VS 2013 workarounds
+#ifdef _MSC_VER
+#if _MSC_VER <= 1800
+
+#pragma pop_macro("GSL_MSVC_HAS_TYPE_DEDUCTION_BUG") 
+#undef GSL_MSVC_HAS_TYPE_DEDUCTION_BUG 
+
```

Comment:
The `#undef` after the pop will remove any previous binding that the push was attempting to preserve.
I think it is better to just remove the push and the pops.